plugin/metadata/metadata.go

```diff
@@ -20,10 +20,15 @@ type Metadata struct {
 // Name implements the Handler interface.
 func (m *Metadata) Name() string { return "metadata" }
 
+// ContextWithMetadata is exported for use by provider tests
```

Comment:
Is this useful enough to add here?

Reply:
Yes, without this you can't do a real unit test, you have to do more of an end-to-end test which is a lot more painful.